src/test/java/com/github/packageurl/PackageURLTest.java

```diff
@@ -210,11 +210,9 @@ public void testConstructorWithInvalidNumberType() throws MalformedPackageURLExc
     }
 
     @Test
-    public void testConstructorWithInvalidSubpath() throws MalformedPackageURLException {
-        exception.expect(MalformedPackageURLException.class);
-
-        PackageURL purl = new PackageURL("pkg:GOLANG/google.golang.org/genproto@abcdedf#invalid/%2F/subpath");
-        Assert.fail("constructor with `invalid/%2F/subpath` should have thrown an error and this line should not be reached");
+    public void testConstructorWithValidSubpathContainingSlashIsDropped() throws MalformedPackageURLException {
+        PackageURL purl = new PackageURL("pkg:GOLANG/google.golang.org/genproto@abcdedf#valid/%2F/subpath");
+        Assert.assertEquals("valid/subpath", purl.getSubpath());
```

Comment:
The previous version seems the correct one: the constructor should throw due to the encoded solidus `/`

Reply:
I made this valid, but now I am trying to remember why. I think the handling of `namespace` and `subpath` may be different.

`namespace`:

> When percent-decoded, a segment:
>> must not contain a `'/'`
>> must not be empty

`subpath`:

> When percent-decoded, a segment: - MUST NOT contain a `'/'` - MUST NOT be any of `'..'` or `'.'` - MUST NOT be empty

But, later:

> Discard empty, '.' and '..' segments

So, are they are allowed or not?

Reply:
@ppkarwasz can you review again with the current test cases that were added to this pull request?

Reply:
See below.